src/expr/node.cpp

```diff
@@ -31,9 +31,21 @@ TypeCheckingExceptionPrivate::TypeCheckingExceptionPrivate(TNode node,
     : Exception(message), d_node(new Node(node))
 {
 #ifdef CVC4_DEBUG
+  std::stringstream ss;
   LastExceptionBuffer* current = LastExceptionBuffer::getCurrent();
   if(current != NULL){
-    current->setContents(toString().c_str());
+    // Since this node is malformed, we cannot use toString().
+    // Instead, we print the kind and the children.
+    ss << message << " "
+       << "node kind: " << node.getKind() << ". children: ";
```

Comment:
You don't need to add `message` here, only the node representation should be added to the stringstream.